sql/builtin_functions.h

```diff
@@ -137,6 +150,13 @@ struct RegisterBuiltin {
                         }
                     };
 
+                    bool constantArgs = determinism == DETERMINISTIC;
+                    for (auto& arg : args) {
+                        constantArgs = constantArgs && arg.metadata.isConstant;
+                    }
+
+                    result.resultMetadata.isConstant = constantArgs;
```

Comment:
Here also, I would put `constantArgs && (determinism == DETERMINISTIC)` and remove it from the above.